packages/muelu/test/scaling/Driver.cpp

```diff
@@ -284,15 +284,29 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib& lib, int ar
   TEUCHOS_TEST_FOR_EXCEPTION(xmlFileName != "" && yamlFileName != "", std::runtime_error,
                              "Cannot provide both xml and yaml input files");
 
+  // Instead of checking each time for rank, create a rank 0 stream
+  RCP<Teuchos::FancyOStream> fancy = Teuchos::fancyOStream(Teuchos::rcpFromRef(std::cout));
+  Teuchos::FancyOStream& out = *fancy;
+  out.setOutputToRootOnly(0);
+ 
   ParameterList paramList;
+  auto inst = xpetraParameters.GetInstantiation();
+  
   if (yamlFileName != "") {
     Teuchos::updateParametersFromYamlFileAndBroadcast(yamlFileName, Teuchos::Ptr<ParameterList>(&paramList), *comm);
-
   } else {
-    xmlFileName = (xmlFileName != "" ? xmlFileName : "scaling.xml");
+    if (inst == Xpetra::COMPLEX_INT_INT)
```

Comment:
How about using `(TYPE_EQUAL(Scalar, std::complex<double>) || TYPE_EQUAL(Scalar, std::complex<float>))`, so that this also works when `GlobalOrdinal!=int`?

Reply:
The MueLu test ETI system probably doesn't work with GlobalOrdinal!=int either (since there's no Xpetra::COMPLEX_INT_LONGLONG )

Reply:
But once we enable having complex and GlobalOrdinal!=int, we will have to change this bit again, no?

Reply:
Correct.  One fix at a time :)